pinot-controller/src/main/java/org/apache/pinot/controller/util/ConsumingSegmentInfoReader.java

```diff
@@ -205,15 +205,19 @@ static public class ConsumingSegmentInfo {
     public long _lastConsumedTimestamp;
     @JsonProperty("partitionToOffsetMap")
     public Map<String, String> _partitionToOffsetMap;
+    @JsonProperty("partitionToOffsetLagMap")
+    public Map<String, String> _partitionToOffsetLagMap;
```

Comment:
it would be helpful here to also return the upstreamLatestOffset. Just offset lag might sometimes not be too indicative given they might not be contiguous? 
Another thought, instead of so many partitionTo* maps now, we could also have a `Map<String, ConsumingSegmentInfoReader.PartitionOffsetInfo> _partitionToOffsetInfo` where this value object could hold current consumer offset, latest stream offset, and lag

Reply:
> it would be helpful here to also return the upstreamLatestOffset. Just offset lag might sometimes not be too indicative given they might not be contiguous?

Sure. Makes sense. 

> Another thought, instead of so many partitionTo* maps now, we could also have a Map<String, ConsumingSegmentInfoReader.PartitionOffsetInfo> _partitionToOffsetInfo where this value object could hold current consumer offset, latest stream offset, and lag 

I tried doing this ☝️ Then, I ran into some json serializer error that I couldn't figure out. Let me give it another try. If I get stuck, I will paste the error and we can work through it.

Reply:
I think I was able to make this change. Please take a look.